src/plugins/discover/public/utils/get_valid_filters.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DataView } from '@kbn/data-views-plugin/common';
+import { Filter } from '@kbn/es-query';
+
+export const getValidFilters = (dataView: DataView, filters: Filter[]): Filter[] => {
+  // We need to disable scripted filters that are not part of this data view
+  // since we can't guarantee they'll succeed for the current data view and
+  // can lead to runtime errors
+  return filters.map((filter) => {
+    return {
+      ...filter,
+      meta: {
+        ...filter.meta,
+        disabled:
+          filter.meta.disabled ||
+          (filter.meta.index !== dataView.id && Boolean(filter.query?.script)),
```

Comment:
This is a great base for the fields of other types in a follow up I'd suggest to make it more granular, theoretically the same field in this dataView could exist, so it should be checked if there's a field like this available before disabling it. What's more it would be great to change ist index of any filter to the new data view id, because it would be even cleaner, and I see no good reason why we should keep the old id in this case (note that this would not also apply to scripted fields)